api/circulation.py

```diff
@@ -342,7 +342,7 @@ def fulfill(self, patron, pin, licensepool, delivery_mechanism, sync_on_failure=
                 )
             else:
                 raise NoActiveLoan(_("Cannot find your active loan for this work."))
-        if loan.fulfillment is not None and loan.fulfillment != delivery_mechanism:
+        if loan.fulfillment is not None and loan.fulfillment != delivery_mechanism and not delivery_mechanism.delivery_mechanism.is_streaming:
```

Comment:
In theory a streaming delivery mechanism could be exclusive of other delivery mechanisms, so we'd have to store that information per data source per mechanism and check if there are any conflicts. Fortunately for now that doesn't seem to be the case.